src/voting-tools-rs/src/logic.rs

```diff
@@ -122,15 +122,29 @@ impl Reg {
         // Translate registration to Cardano metadata type so we can serialize it correctly
         let mut meta_map: MetadataMap = MetadataMap::new();
         let delegations = match self.metadata.delegations.clone() {
-            Delegations::Delegated(_) => {
-                TransactionMetadatum::new_text("foo".to_string()).map_err(|_| eyre!("uh oh"))?
-                // TODO: this seems weird
+            Delegations::Delegated(delegations) => {
+                let mut metadata_list = MetadataList::new();
+                for (delegation, weight) in delegations {
+                    let mut inner_metadata_list = MetadataList::new();
+                    inner_metadata_list.add(
+                        &TransactionMetadatum::new_bytes(hex::decode(
+                            delegation.trim_start_matches("0x"),
+                        )?)
+                        .unwrap(),
```

Comment:
How confident are we that this panic is impossible to reach?

Reply:
Ok, adding error handling